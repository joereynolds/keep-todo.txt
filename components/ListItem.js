import React from 'react';
import { StyleSheet, View } from 'react-native';
import {CheckBox} from 'react-native-elements';

const ListItem = (props) => {
    return (
        <View>
            <CheckBox
                title={props.title}
                checked={props.checked}
                checkedIcon="check-square"
                checkedColor="gray"
                containerStyle={styles.container}
                textStyle={props.checked === true ? styles.textChecked : styles.textUnchecked}
            />
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        marginLeft: 0,
        marginRight: 0,
    },
    textChecked: {
            textDecorationLine: 'line-through'
    },
    textUnchecked: {
        textDecorationLine: 'none'
    }
});

export default ListItem;
