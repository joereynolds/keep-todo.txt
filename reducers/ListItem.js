export default function listItemReducer(state, action) {
    if (action.type === 'todo/item/complete') {
        return {
            ...state
            // TODO - 'complete' the selected item
        }
    }

    return state;
}
