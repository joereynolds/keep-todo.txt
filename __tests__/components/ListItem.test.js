import React from 'react';
import renderer from 'react-test-renderer';
import ListItem from '../../components/ListItem';

describe('<ListItem />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ListItem title='this-is-my-title'/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
