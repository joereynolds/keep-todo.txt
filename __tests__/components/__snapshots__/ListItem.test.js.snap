// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> renders correctly 1`] = `
<View>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fafafa",
        "borderColor": "#ededed",
        "borderRadius": 3,
        "borderWidth": 1,
        "margin": 5,
        "marginLeft": 0,
        "marginRight": 0,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "#43484d",
            "fontWeight": "bold",
            "marginLeft": 10,
            "marginRight": 10,
            "textDecorationLine": "none",
          }
        }
        testID="checkboxTitle"
      >
        this-is-my-title
      </Text>
    </View>
  </View>
</View>
`;
